ffi/framebuffer_linux.lua

```diff
@@ -36,9 +36,12 @@ function framebuffer:init()
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
     -- Newer eink framebuffer (Kindle Touch, Paperwhite, Kobo)
     elseif ffi.string(finfo.id, 11) == "mxc_epdc_fb" then
-        -- it seems that finfo.smem_len is unreliable on kobo
         -- Figure out the size of the screen in bytes
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
+        -- Kobo hasn't updated smem_len when the depth was changed from 16 to 32
+        if self.fb_size > finfo.smem_len then
```

Comment:
`finfo.smem_len` is accurate in 16-bit mode but is half of what is should be in 32-bit mode.

If `self.fb_size` is greater that `finfo.smem_len` then writing to the `mmap`ed region will generate a segfault.

I agree with the more specific test you suggested and I've created #486 (because I forgot to create a branch in this PR).